src/ui/App.js

```diff
@@ -1,47 +1,85 @@
 import Api from "./Api.js";
+import { Deck, DeckExpiredError } from "./Deck.js";
 import Hand from "./Hand.js";
-import { useState, useCallback } from "react";
+import { useState, useCallback, useEffect } from "react";
 
 /**
    Note: For each game, we'll only allow the "Draw" to happen once.
    This means we should hide the "Draw" button after it is clicked.
    Better yet, let's change it to a different "Play Again" button that
    resets everything and plays another hand (with a new Deck).
 **/
-export default function App({ initialCards }) {
+export default function App({ initialCards, initialDeck, initialEnded }) {
   const [cards, setCards] = useState(initialCards);
   const [selected, setSelected] = useState([]);
+  const [deck, setDeck] = useState(initialDeck);
+  const [dealt, setDealt] = useState(initialEnded);
+  const [error, setError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState("");
+  console.log("initialDeck: " + initialDeck);
+  console.log("current deck: " + deck);
+  console.log("setDeck? " + setDeck);
 
-  function toggleSelected(index) {
-    if (!selected.includes(index)) {
-      setSelected(selected.concat([index]));
-    } else {
-      setSelected(selected.filter((elt) => elt !== index));
+  useEffect(() => {
```

Comment:
Generally, `useEffect` makes me nervous since it's too easy to abuse. I'm a bit unclear what w'ere doing here, but it seems to be related to error checking. Is this when there's an error on the server or an error on the client?